src/logger.ts

```diff
@@ -0,0 +1,135 @@
+// Determine if structured logs are supported (node >= 10). If something goes wrong,
+import { format } from 'util';
+
+// safely preserve unpatched console.* methods in case of compat require
+const unpatchedConsole = {
```

Comment:
It does work, because the first thing `compat` does is require `logger` which preserves the unpatched console methods. Added a comment to clarify.

This wasn't in my original proposal but Kato asked if it was possible and it turned out to be easier than expected. I do worry about the chance for bugs here but this is just one way to use the SDK and provides a convenience. Anyone who encounters bugs we can just tell to use the other ways of using the SDK.

Reply:
Ah yes, compat does import logger first, this is really subtle behavior, it's good you added a comment to clarify. 

I'm still not a fan of `compat` and I think it's just making this SDK hard to maintain (it's already way too complicated and has too many subtle behaviors) But if you feel like the risk is worth the user experience benefit, we can keep it in.